nav2_ros_common/include/nav2_ros_common/validate_messages.hpp

```diff
@@ -56,6 +56,8 @@ bool validateMsg(const double & num)
   return true;
 }
 
+const double MAX_COVARIANCE = 1e6;
```

Comment:
You could keep a minimum, these do have to be positive valued. So you can check if the element is `< 0.0`. That would be the most complete solution to your needs